File: ejercicios hechos/Python/estructuras alternativas/act18.py
#Programa: act18.py
#Propósito: Realiza un programa que pida el día de la semana (del 1 al 7) y escriba el día correspondiente. Si introducimos otro número nos da un error.
#Autor: Jose Manuel Serrano Palomo.
#Fecha: 17/10/2019
#
#Variables a usar:
# dia el valor número asignado al día de la semana
# dia_semana será un array que contenga los valores
#
#Algoritmo:
# LEER dia
# comprobar si es un número del 1 al 7 en caso contrario mensaje de error
# si es 1 mostrar lunes
# si es 2 mostrar martes
# si es 3 mostrar miercoles
# si es 4 mostrar jueves
# si es 5 mostrar viernes
# si es 6 mostrar sabado
# si es 7 mostrat domingo

print("Muestra el día de la semana")
print("----------------------------\n")

# Leemos los datos
dia = int(input("Introduce el número del día de la semana: "))
dia_semana = ["Lunes","Martes","Miercoles","Jueves","Viernes","Sabado","Domingo"]

#comprobamos los datos
if dia < 1 or dia > 7:
    print("introduce un valor entre 1 y 7")
else:
    if dia == 1:
        print(f"El día de la semana es {dia_semana[dia-1]}")
        
    else:
        if dia == 2:
            print(f"El día de la semana es {dia_semana[dia-1]}")
        
        else:
            if dia == 3:
                print(f"El día de la semana es {dia_semana[dia-1]}")
        
            else:
                if dia == 4:
                    print(f"El día de la semana es {dia_semana[dia-1]}")
        
                else:
                    if dia == 5:
                        print(f"El día de la semana es {dia_semana[dia-1]}")
        
                    else:
                        if dia == 6:
                            print(f"El día de la semana es {dia_semana[dia-1]}")
        
                        else:
                            if dia == 7:
                                print(f"El día de la semana es {dia_semana[dia-1]}")

File: ejercicios terminados/Java/estructuras alternativas/act18.java
/*Programa: act18.java
 * 
 *Propósito: Realiza un programa que pida el día de la semana (del 1 al 7) y escriba el día correspondiente. Si introducimos otro número nos da un error.
 * 
 *Autor: Jose Manuel Serrano Palomo.
 * 
 *Fecha: 09/11/2019
 *
 *Análisis:
 * Leemos un día de la semana como número y lo mostramos escrito en pantalla y si hay un valor erroneo mensaje de error
 * 
 *Diseño:
 * 1. Leer dia de la semana
 * 2. Según el valor mostrar el nombre
 * 3. si el dia es incorrecto mensaje de error
*/
import java.util.Scanner;
public class act18 {
    public static void main(String[] args) {
        System.out.println("Programa para saber el dia de la semana");
        System.out.println("---------------------------------------");
        Scanner s = new Scanner(System.in);
        int dia;

        System.out.print("Dime un día de la semana (1-7): ");
        dia = s.nextInt();

        switch (dia) {
          case 1:
            System.out.println("Lunes");
            break;
          case 2:
            System.out.println("Martes");
            break;
          case 3:
            System.out.println("Miércoles");
            break;
          case 4:
            System.out.println("Jueves");
            break;
          case 5:
            System.out.println("Viernes");
            break;
          case 6:
            System.out.println("Sábado");
            break;
          case 7:
            System.out.println("Domingo");
            break;
          default:
            System.out.println("Día incorrecto");
        }
    }
}
